src/sentry/integrations/jira/webhooks/installed.py

```diff
@@ -35,14 +40,27 @@ def post(self, request: Request, *args, **kwargs) -> Response:
             provider_key=self.provider,
         ).capture() as lifecycle:
             token = self.get_token(request)
-
             state = request.data
             if not state:
                 lifecycle.record_failure(ProjectManagementFailuresReason.INSTALLATION_STATE_MISSING)
                 return self.respond(status=status.HTTP_400_BAD_REQUEST)
 
             key_id = jwt.peek_header(token).get("kid")
+            lifecycle.add_extras(
+                {
+                    "key_id": key_id,
+                    "base_url": state.get("baseUrl", ""),
+                    "description": state.get("description", ""),
+                    "clientKey": state.get("clientKey", ""),
+                }
+            )
+
             if key_id:
+                if key_id in INVALID_KEY_IDS:
+                    lifecycle.record_halt(halt_reason="JWT contained invalid key_id (kid)")
+                    return self.respond(
+                        {"detail": "Invalid key id"}, status=status.HTTP_400_BAD_REQUEST
+                    )
```

Comment:
i wonder if instead of this, we have a try/except to verify the claims. if we encounter a ValueError while decoding AND the key is in invalid ids, we should record halt

the reason being we should still allow the decoding to happen as that is the expected behavior, we just don't want the failure to be counted as a failure

Reply:
Yeah this was another option I talked with @GabeVillalobos about, but we went this method as it was easiest and the installation webhook module will get rewritten soon(TM) as Jira Forge uses a different authentication method/doesn't need this JWT verification part.